tests/test_livysession.py

```diff
@@ -39,11 +36,18 @@ class TestLivySession(object):
     log_json = json.loads('{"id":6,"from":0,"total":212,"log":["hi","hi"]}')
 
     def __init__(self):
+        self.http_client = None
```

Comment:
http_client and spark_events used by almost all test cases. Moving them to test fixture makes sense.
They are recreated before each test case is executed. There's no coupling between test cases.